### PR TITLE
feat(odyssey-react): adds tab component

### PR DESCRIPTION
**Pull request by arnoldsandoval-okta**

### CODE REVIEW

File: packages/odyssey-react/src/components/Tab/Tab.test.tsx [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+      <Tabs id={id} ariaLabel={ariaLabel} onTabChange={handleTabChange}>
+        <Tabs.Panel id="sb-tabs-example-1" label="Tab 1">TabPanel 1</Tabs.Panel>
+        <Tabs.Panel id="sb-tabs-example-2" label="Tab 2">TabPanel 2</Tabs.Panel>
+        <Tabs.Panel id="sb-tabs-example-3" label="Tab 3">TabPanel 3</Tabs.Panel>
+      </Tabs>
+    );
+
+    fireEvent.click(getByText('Tab 2'));
+
+    expect(handleTabChange).toHaveBeenCalledTimes(1);
```
Comment (jeffbelser-okta) [inline packages/odyssey-react/src/components/Tab/Tab.test.tsx:82]: consider adding an assertion here for the parameters of the callback too, we provide the `newSelectedId` correct?

File: packages/odyssey-react/src/components/Tab/index.tsx [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+ */
+
+import React, { useState } from "react";
+import type { FunctionComponent, ReactElement, ReactText } from "react";
+
+export type PropsTabs = {
+  /**
+   * A collection of Tab.Panels to be rendered
+   */
+  children: ReactElement[],
```
Comment (jeffbelser-okta) [inline packages/odyssey-react/src/components/Tab/index.tsx:20]: \o/

File: packages/odyssey-react/src/components/Tab/index.tsx [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+  const handleTabChange = (newSelectedId: string) => {
+    setSelectedTabId(newSelectedId)
+
+    if (onTabChange) {
+      onTabChange(newSelectedId);
+    }
+  }
+
+  return (
+    <Tabs.Container id={id} ariaLabel={ariaLabel}>
```
Comment (jeffbelser-okta) [inline packages/odyssey-react/src/components/Tab/index.tsx:108]: nice I like having the `Container` components rendered internally to an abstracted `Tab` and `Tab.Panel` public interface

File: packages/odyssey-react/src/components/Tab/index.tsx [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+}
+
+export type PropsTabsPanelContainer = { 
+  children: ReactElement | ReactElement[]
+}
+
+export type PropsTabsPanel = { 
+  children: ReactText | ReactElement | ReactElement[],
+  label: string,
+  id?: string,
```
Comment (jeffbelser-okta) [inline packages/odyssey-react/src/components/Tab/index.tsx:57]: hmmm this is tricky. If `id` is optional we break `a11y` and the use of `arial-labelledby` below.

one the other hand if we autogenerate IDs to solve for this we make the `defaultSelected` API unworkable 😅  We may need to revisit this to use index position or a declarative `isSelected` API

File: packages/odyssey-react/src/components/Tab/index.tsx [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+            ariaControls={id}
+            selected={id === selectedTabId}
+            onClick={() => handleTabChange(id)}
+          >
+            {label}
+          </Tabs.Tab>
+        ))}
+      </Tabs.List>
+      <Tabs.PanelContainer>
+        {children.map(({ props: { label, id, children: tabPabelChildren } }) => (
```
Comment (jeffbelser-okta) [inline packages/odyssey-react/src/components/Tab/index.tsx:123]: consider mapping through children only once building an array of tuples (or whatever DS) of these components to render here and above on 110.

File: packages/odyssey-react/src/components/Tab/index.tsx [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+    </button>
+)
+
+Tabs.PanelContainer = ({ children }) => (<div className="ods-tabs--tabpanel">{children}</div>)
+
+Tabs.Panel = ({ children, id, selected }) => (
+  <div
+    id={id}
+    role="tabpanel"
+    aria-labelledby={id + '-tab'}
```
Comment (jeffbelser-okta) [inline packages/odyssey-react/src/components/Tab/index.tsx:168]: see note above about type annotations and a11y

File: packages/odyssey-react/src/components/Tab/Tab.test.tsx [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+        <Tabs.Panel id="sb-tabs-example-2" label="Tab 2">TabPanel 2</Tabs.Panel>
+        <Tabs.Panel id="sb-tabs-example-3" label="Tab 3">TabPanel 3</Tabs.Panel>
+      </Tabs>
+    );
+    
+    fireEvent.click(getByText('Tab 3'));
+
+    expect(getByText('TabPanel 3')).toBeVisible();
+    expect(getByText('Tab 3')).toHaveAttribute('aria-selected', 'true');
+    expect(getByText('Tab 1')).toHaveAttribute('aria-selected', 'false');
```
Comment (jeffbelser-okta) [inline packages/odyssey-react/src/components/Tab/Tab.test.tsx:65]: consider using `getAllByRole('tab', { selected: true })` here. You can assert the result is one.

**Review by jeffbelser-okta — APPROVED** [review]
strong work 💪  We have some a11y gaps for keyboard nav and focus which won't fully satisfy AC in JIRA but this is a great place to keep moving forward from!

**Comment by arnoldsandoval-okta** [comment]
Will update tests in follow up PR

**Pull request merged**